cpp/daal/src/algorithms/objective_function/cross_entropy_loss/cross_entropy_loss_dense_default_batch_impl.i

```diff
@@ -524,8 +534,11 @@ services::Status CrossEntropyLossKernel<algorithmFPType, method, cpu>::doCompute
             const algorithmFPType interceptFactor = (parameter->interceptFlag ? 1 : 0);
             const auto hSize                      = nBeta * nBeta;
             TlsSum<algorithmFPType, cpu> tlsData(hSize);
-
+            SafeStatus safeStat;
             daal::threader_for(n, n, [&](size_t i) {
+                ReadRows<algorithmFPType, cpu> xr(const_cast<NumericTable *>(dataNT), i, 1);
```

Comment:
A big degradation can happen here. Each line in parallel is converted, not processed by the block. SOA tables will be very expensive here.